app/reconciliation.go

```diff
@@ -0,0 +1,116 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bnb-chain/node/common"
+	"github.com/bnb-chain/node/common/types"
+)
+
+const globalAccountNumber = "globalAccountNumber"
+
+// reconBalance will do reconciliation for accounts balances.
+func (app *BinanceChain) reconBalance(ctx sdk.Context) {
+	currentHeight := ctx.BlockHeight()
+
+	accountStore, ok := app.GetCommitMultiStore().GetCommitStore(common.AccountStoreKey).(*store.IavlStore)
+	if !ok {
+		panic("cannot convert account store to ival store")
+	}
+	accPre, accCurrent := app.getAccountChanges(ctx, accountStore)
+	accountStore.ResetDiff()
+
+	tokenStore, ok := app.GetCommitMultiStore().GetCommitStore(common.TokenStoreKey).(*store.IavlStore)
+	if !ok {
+		panic("cannot convert token store to ival store")
+	}
+	tokenPre, tokenCurrent := app.getTokenChanges(ctx, tokenStore)
+	tokenStore.ResetDiff()
+
+	left := tokenPre.Plus(accCurrent)
+	right := tokenCurrent.Plus(accPre)
+
+	if !left.IsEqual(right) {
+		err := fmt.Sprintf("unbalanced at block %d, pre: %s, current: %s \n",
+			currentHeight, left.String(), right.String())
+		ctx.Logger().Error(err)
+		panic(err)
+	}
+}
+
+func (app *BinanceChain) getAccountChanges(ctx sdk.Context, accountStore *store.IavlStore) (sdk.Coins, sdk.Coins) {
+	preCoins := sdk.Coins{}
+	currentCoins := sdk.Coins{}
+
+	diff := accountStore.GetDiff()
+	version := accountStore.GetTree().Version() - 1
```

Comment:
why minus by `1`?

Reply:
It will cause we skip one block verification and cause a consensus issue.

Reply:
Refactor the code - the unbalanced height will be saved. It is also useful to bypass the panic when we want to resume the blockchain.